commit 448d25759ea7ee0f680c7343c87deb4dd6ebf799
Author: Eric Anholt <eric@anholt.net>
Date:   Wed Sep 20 15:59:34 2017 -0700

    drm/vc4: Reject HDMI modes with too high of clocks.
    
    Peter Robinson reported issues on Fedora with 4k monitors not having
    their modes filtered down to 1920x1080 on Raspberry Pi.
    
    v2: Fix vc5 typo in place of vc4.
    
    Cc: Peter Robinson <pbrobinson@redhat.com>
    Signed-off-by: Eric Anholt <eric@anholt.net>
    Link: https://patchwork.freedesktop.org/patch/msgid/20170920225935.14566-1-eric@anholt.net
    Acked-by: Daniel Vetter <daniel.vetter@ffwll.ch> (v1)
    (cherry picked from commit 32e823c63e90f7535ea1cc5311d25c0233e1456d)

diff --git a/drivers/gpu/drm/vc4/vc4_hdmi.c b/drivers/gpu/drm/vc4/vc4_hdmi.c
index 0b2088264039..8ab20c1e9e69 100644
--- a/drivers/gpu/drm/vc4/vc4_hdmi.c
+++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
@@ -695,7 +695,22 @@ static void vc4_hdmi_encoder_enable(struct drm_encoder *encoder)
 	}
 }
 
+static enum drm_mode_status
+vc4_hdmi_encoder_mode_valid(struct drm_encoder *crtc,
+			    const struct drm_display_mode *mode)
+{
+	/* HSM clock must be 108% of the pixel clock.  Additionally,
+	 * the AXI clock needs to be at least 25% of pixel clock, but
+	 * HSM ends up being the limiting factor.
+	 */
+	if (mode->clock > HSM_CLOCK_FREQ / (1000 * 108 / 100))
+		return MODE_CLOCK_HIGH;
+
+	return MODE_OK;
+}
+
 static const struct drm_encoder_helper_funcs vc4_hdmi_encoder_helper_funcs = {
+	.mode_valid = vc4_hdmi_encoder_mode_valid,
 	.disable = vc4_hdmi_encoder_disable,
 	.enable = vc4_hdmi_encoder_enable,
 };
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1183_448d25759ea7ee0f680c7343c87deb4dd6ebf799.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1183_448d25759ea7ee0f680c7343c87deb4dd6ebf799.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
